vector/src/main/java/im/vector/activity/MediasPreviewerActivity.java

```diff
@@ -0,0 +1,127 @@
+package im.vector.activity;
+
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.app.Activity;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+import android.webkit.WebView;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.VideoView;
+
+import org.matrix.androidsdk.data.RoomMediaMessage;
+import org.matrix.androidsdk.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import im.vector.R;
+import im.vector.adapters.MediasPreviewAdapter;
+
+public class MediasPreviewerActivity extends MXCActionBarActivity {
+
+    /**
+     * the picture uri if a picture is taken with the camera
+     */
+    public static final String EXTRA_CAMERA_PICTURE_URI = "EXTRA_CAMERA_PICTURE_URI";
+
+    private static final String LOG_TAG = MediasPreviewerActivity.class.getSimpleName();
+
+    @Override
+    public void initUiAndData() {
+
+        if (CommonActivityUtils.shouldRestartApp(this)) {
+            Log.d(LOG_TAG, "onCreate : restart the application");
+            CommonActivityUtils.restartApp(this);
+            return;
+        }
+
+        if (CommonActivityUtils.isGoingToSplash(this)) {
+            Log.d(LOG_TAG, "onCreate : Going to splash screen");
+            return;
+        }
+
+        Toolbar toolbar = (android.support.v7.widget.Toolbar) findViewById(R.id.room_toolbar);
+
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
```

Comment:
Replace the 3 lines above by configureToolbar() and rename the id of the toolbar in layout to `toolbar`